### PR TITLE
AB#3557 - add view-playload feature

### PR DESCRIPTION
**Pull request by leiye29**
### Description

1. display playload data for tester
2. set feature flag for view-playload

### Screenshots (if applicable)

image

### Checklist

- [x] I have tested the changes locally
- [ ] I have updated the documentation if necessary
- [ ] I have added/updated tests that prove my fix is effective or that my feature works
- [x] I have checked that my code follows the project's coding style by running `npm run format:check`
- [x] I have checked that my code contains no linting errors by running `npm run lint`
- [x] I have checked that my code contains no type errors by running `npm run typecheck`
- [x] I have checked that all unit tests pass by running `npm run test:unit -- run`
- [x] I have checked that all e2e tests pass by running `npm run test:e2e`

### CODE REVIEW

File: frontend/app/routes/$lang+/_public+/apply+/$id+/adult/review-information.tsx [hunk 1]
```diff
@@ -495,6 +520,15 @@ export default function ReviewInformation() {
...
       </div>
+      {useFeature('view-playload') && (
+        <div className="mt-8">
+          <h1 className="text-x1 mb-4 font-semibold">JSON Data</h1>
+          <pre id="playload" className="rounded-log block max-h-96 overflow-auto bg-gray-100 p-4 shadow-inner">
+            {JSON.stringify(playload, null, 2)}
+          </pre>
+          <Button onClick={copyPlayLoadHandler}>copy</Button>
+        </div>
+      )}
```
Comment (gregory-j-baker) [inline frontend/app/routes/$lang+/_public+/apply+/$id+/adult/review-information.tsx:531]: This should be a reusable component that we can drop onto any page.. something like this:

``` tsx
<Debug enableCopy data={payload} />
```

Where `enableCopy` enables/disables the copy button, and `payload` is the raw data that you want to display (the component would `JSON.stringify()` it). This component should also render conditionally, depending on the feature flag.

Something like this:

``` tsx
function Debug({ data, enableCopy = true }) {
  if (featureFlagEnabled) {
    const json = JSON.stringify(data, null, 2);
    const onCopyClick = navigator.clipboard.writeText(json);
    return (
      <>
        <pre>...</pre>
        {enableCopy && <Button>...</Button>}
      </>
    );
  }
}
```

File: frontend/.env.example [hunk 2]
```diff
@@ -50,7 +50,7 @@ AUTH_RASCL_LOGOUT_URL=http://localhost:3000/
 
 # Application feature flags -- see ./app/utils/env.server.ts for valid values
 # (optional; default: "")
-ENABLED_FEATURES=doc-upload,email-alerts,hcaptcha,view-personal-info,view-applications,view-letters,view-messages,edit-personal-info,status,authenticated-status-check, update-governmental-benefit,dependent-status-checker
+ENABLED_FEATURES=doc-upload,email-alerts,hcaptcha,view-personal-info,view-applications,view-letters,view-messages,edit-personal-info,status,authenticated-status-check, update-governmental-benefit,dependent-status-checker,view-playload
```
Comment (gregory-j-baker) [inline frontend/.env.example:53]: I know this isn't a part of your PR, but can you remove the space before `update-governmental-benefit`, please. I don't think it should be there.

**Review by gregory-j-baker — CHANGES_REQUESTED** [review]

**Comment by gregory-j-baker** [comment]
What do you think about formatting the html to look something like this (except with the correct light theme instead of dark)?

image

**Comment by gregory-j-baker** [comment]
I also just realized that github has a copy button on its comments: 

image

File: frontend/app/routes/$lang+/_public+/apply+/$id+/adult/review-information.tsx (hunk 1 above)
Comment (leiye29) [reply frontend/app/routes/$lang+/_public+/apply+/$id+/adult/review-information.tsx]: new component debug-payload

File: frontend/.env.example (hunk 2 above)
Comment (leiye29) [reply frontend/.env.example]: removed

File: frontend/app/routes/$lang+/_public+/apply+/$id+/adult/review-information.tsx [hunk 3]
```diff
@@ -217,6 +230,7 @@ export async function action({ context: { session }, params, request }: ActionFu
   }
 
   // TODO submit to the API and grab the confirmation code from the response
+  // any field change here will update playload assignment in loader()
```
Comment (sebastien-comeau) [inline frontend/app/routes/$lang+/_public+/apply+/$id+/adult/review-information.tsx:233]: `playload` to `payload`

**Review by sebastien-comeau — CHANGES_REQUESTED** [review]
Rename `payload`

File: frontend/app/components/debug-payload.tsx [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+export function DebugPayload({ data, enableCopy }: debugPayloadtProps) {
+  const json = JSON.stringify(data, null, 2);
+  return (
+    <>
+      <pre className="rounded-log block max-h-96 overflow-auto bg-gray-100 p-4 shadow-inner">{json}</pre>
+      {enableCopy && <Button onClick={() => navigator.clipboard.writeText(json)}>copy</Button>}
+    </>
+  );
+}
```
Comment (sebastien-comeau) [inline frontend/app/components/debug-payload.tsx:16]:
```suggestion
import { useEffect, useState } from 'react';

import { faCheck, faClipboard } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

import { Button } from './buttons';

export type debugPayloadtProps = {
  data?: unknown;
  enableCopy?: boolean;
};

export function DebugPayload({ data, enableCopy }: debugPayloadtProps) {
  const [hasCopied, setHasCopied] = useState(false);

  useEffect(() => {
    setTimeout(() => {
      setHasCopied(false);
    }, 2000);
  }, [hasCopied]);

  const json = JSON.stringify(data, null, 2);

  const copyToClipboard = () => {
    navigator.clipboard.writeText(json);
    setHasCopied(true);
  };

  return (
    <>
      <pre className="rounded-log block max-h-96 overflow-auto bg-gray-100 p-4 shadow-inner">{json}</pre>
      {enableCopy && (
        <Button onClick={copyToClipboard}>
          Copy
          <FontAwesomeIcon icon={hasCopied ? faCheck : faClipboard} className="ms-3 block size-4" />
        </Button>
      )}
    </>
  );
}
```

**Review by sebastien-comeau — CHANGES_REQUESTED** [review]
This suggestion adds an icon to give feedback when the content is copied.

File: frontend/app/routes/$lang+/_public+/apply+/$id+/adult/review-information.tsx (hunk 3 above)
Comment (leiye29) [reply frontend/app/routes/$lang+/_public+/apply+/$id+/adult/review-information.tsx]: fixed

File: frontend/app/components/debug-payload.tsx (hunk 4 above)
Comment (leiye29) [reply frontend/app/components/debug-payload.tsx]: updated

**Review by sebastien-comeau — APPROVED** [review]

File: frontend/app/components/debug-payload.tsx [hunk 5]
```diff
@@ -0,0 +1,35 @@
+import { useEffect, useState } from 'react';
+
+import { faCheck, faClipboard } from '@fortawesome/free-solid-svg-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+
+import { Button } from './buttons';
+
+export type debugPayloadtProps = {
```
Comment (gregory-j-baker) [inline frontend/app/components/debug-payload.tsx:8]: I know this has already been merged.. but there's a typo here.

Comment (sebastien-comeau) [reply frontend/app/components/debug-payload.tsx:8]: Good catch, can be fixed.

**Pull request merged**